packages/cli/src/controller/generate-controller.ts

```diff
@@ -0,0 +1,309 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import fs from 'fs';
+import path from 'path';
+import {FunctionFragment, EventFragment, ConstructorFragment, Fragment} from '@ethersproject/abi/src.ts/fragments';
+import {loadFromJsonOrYaml} from '@subql/common';
+import {
+  EthereumDatasourceKind,
+  EthereumHandlerKind,
+  EthereumTransactionFilter,
+  SubqlRuntimeHandler,
+} from '@subql/common-ethereum';
+import {SubqlRuntimeDatasource as EthereumDs, EthereumLogFilter} from '@subql/types-ethereum';
+import chalk from 'chalk';
+import ejs from 'ejs';
+import {Interface} from 'ethers/lib/utils';
+import * as inquirer from 'inquirer';
+import {upperFirst, difference} from 'lodash';
+import {parseContractPath} from 'typechain';
+import {Document, parseDocument, YAMLSeq} from 'yaml';
+import {SelectedMethod, UserInput} from '../commands/codegen/generate';
+
+interface HandlerPropType {
+  name: string;
+  argName: string;
+  argType: string;
+}
+
+interface AbiPropType {
+  name: string;
+  handlers: HandlerPropType[];
+}
+
+const SCAFFOLD_HANDLER_TEMPLATE_PATH = path.resolve(__dirname, '../template/scaffold-handlers.ts.ejs');
+const ROOT_MAPPING_DIR = 'src/mappings';
+const DEFAULT_HANDLER_BUILD_PATH = './dist/index.js';
+
+export function removeKeyword(inputString: string): string {
+  const removeString = inputString.startsWith('event ') ? 'event ' : 'function ';
+  if (inputString.startsWith(removeString)) {
+    return inputString.slice(removeString.length);
+  }
+  return inputString;
```

Comment:
Some regex for you. I havent fully tested this though
```suggestion
  return inputString.replace(/^(event|function) /, '');
```